### ISSUE TITLE
P256 ciphersuite issue

### ISSUE THREAD

**Issue by raphaelrobert** [raphaelrobert/maelstrom#162]
To be investigated, right now we get the following error in the group tests:

`thread 'group_operations' panicked at 'called Result::unwrap() on an Err value: InvalidPoint', /Users/raph/.cargo/git/checkouts/hpke-rs-b1873fcfc2d2292e/fb79bbd/src/dh_kem.rs:32:50`

**Comment by franziskuskiefer** [comment]
Can you create a minimal example for this and put it an issue in https://github.com/franziskuskiefer/hpke-rs/issues?

**Comment by raphaelrobert** [comment]
Example:

```
let hpke = Hpke::new(
        Mode::Base,
        KemMode::DhKemP256,
        HpkeKdfMode::HkdfSha256,
        HpkeAeadMode::AesGcm128,
    );
    let kp = hpke.generate_key_pair();
    hpke.seal(
        kp.get_public_key_ref(),
        &[],
        &[],
        &[1, 2, 3],
        None,
        None,
        None,
    )
    .unwrap();
```

While the error occurs in `hpke-rs`, I'm not sure what causes it. It could also be in the `ciphersuite` module.

**Comment by franziskuskiefer** [comment]
Ah, I know what's going on. Not an issue in here. Strictly speaking a bug in the evercrypt bindings. It should be fixed on the [0.0.3 branch](https://github.com/franziskuskiefer/evercrypt-rust/tree/0.0.3) branch ([0.0.2](https://github.com/franziskuskiefer/hpke-rs/tree/0.0.2) for hpke). To pull these versions in a little more work is needed though. I filed #167 for this.
